cmd/server/main.go

```diff
@@ -38,97 +36,77 @@ const (
 
 func main() {
 	logger := logrus.New()
-	// create new tcp listenerf
-	ln, err := net.Listen("tcp", ServerAddress)
-	if err != nil {
-		logger.Fatalln(err)
-	}
-
-	interceptors := []grpc.UnaryServerInterceptor{
-		// validation interceptor
-		grpc_validator.UnaryServerInterceptor(),
-		// logger interceptor
-		grpc_logrus.UnaryServerInterceptor(logrus.NewEntry(logger)),
-		// Request-Id interceptor
-		requestid.UnaryServerInterceptor(),
-	}
-	// add authorization interceptor if authz service address is provided
-	if AuthzAddr != "" {
-		interceptors = append(interceptors,
-			// authorization interceptor
-			toolkit_auth.UnaryServerInterceptor(AuthzAddr, applicationID),
-		)
-	}
-	middleware := grpc_middleware.ChainUnaryServer(interceptors...)
-	// create new gRPC server with middleware chain
-	server := grpc.NewServer(
-		grpc.UnaryInterceptor(
-			middleware,
-		),
-	)
 
-	healthChecker := health.NewChecksHandler("/healthz", "/ready")
-	healthChecker.AddReadiness("DB ready check", dbReady)
-	go http.ListenAndServe(HealthAddress, healthChecker.Handler())
-
-	// waiting for database is available.
-	logger.Info("Connecting to database...")
-	dbCheckContext, cancel := context.WithTimeout(context.Background(), time.Minute*5)
-	for {
-		select {
-		case <-time.After(time.Second * 3):
-			if err = dbReady(); err != nil {
-				continue
-			}
-			cancel()
-		case <-dbCheckContext.Done():
-			cancel()
-		}
-		break
-	}
-	if err != nil {
-		logger.Fatalln(err)
-	}
-	logger.Info("Connected to database successfully.")
+	// create new postgres database
 	db, err := gorm.Open("postgres", DBConnectionString)
 	if err != nil {
 		logger.Fatalln(err)
 	}
 
-	// NOTE: Using db.AutoMigrate is a temporary measure to structure the contacts
-	// database schema. The atlas-app-toolkit team will come up with a better
-	// solution that uses database migration files.
-	if err := db.AutoMigrate(&pb.ProfileORM{}, &pb.GroupORM{}, &pb.ContactORM{}, &pb.AddressORM{}, &pb.EmailORM{}).Error; err != nil {
+	grpcServer, err := NewGRPCServer(logger, db)
+	if err != nil {
 		logger.Fatalln(err)
 	}
-	ps, err := svc.NewProfilesServer(db)
+
+	healthChecker := health.NewChecksHandler("/healthz/", "/ready/")
```

Comment:
I see that previously we used `health.NewChecksHandler("/healthz", "/ready")`. Just wonder why this way is used and if it can cause any issues.

I suppose that with current implementation + go 1.9.6 request to `/health` will work fine, while if we change to go 1.10 is may stop working. In go 1.9.6 there is a special handling for trailing "/" (it will insert an implicit permanent redirect for /health), while in go 1.10 this logic is removed. Please take a look at two links below (`Handle` method implementation):
1.9.6 - https://github.com/golang/go/blob/go1.9.6/src/net/http/server.go#L2286
1.10.2 - https://github.com/golang/go/blob/go1.10.2/src/net/http/server.go#L2342

If my assumption is correct than I think it worth to revert it back to using `/health` instead of `/health/`